src/main/java/uk/ac/ebi/tsc/tesk/service/TesServiceImpl.java

```diff
@@ -72,7 +82,14 @@ public TesCreateTaskResponse createTask(TesTask task, User user) {
     @Override
     public TesTask getTask(String taskId, TaskView view, User user) {
 
+
         V1Job taskMasterJob = this.kubernetesClientWrapper.readTaskmasterJob(taskId);
+        if(authorisationProperties.isIgnoreGroupMembership()){
+            if(!taskMasterJob.getMetadata().getLabels().get(LABEL_USERID_KEY).equals(user.getUsername())){
+                logger.info("The task doesn't belong to the user");
+                throw new TaskNotFoundException(taskId);
```

Comment:
I believe the current behaviour in case of trying to see someone else's task is 403. If that is good, is a different story - as it reveals the existence of the task, but if we need a change, that needs to be consistent,
Is there any problem with following the current pattern?